### DISCUSSION TITLE
[Feature] Monitor main library local file changes

### DISCUSSION THREAD

**Discussion by ArtemSmaznov** [immich-app/immich#13074] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I am wondering if it is possible to have a job that would scan the main library for any local file changes and update records in the db accordingly?
This would also resolve the request to update file names (in a way) mentioned in #1775 

use case 1:
syncing the main photos dir from main machine to the server so that I can work on it locally (e.g. generate jpegs from raw) and not having to worry about remembering to re-upload every change to immich as you can just rely on a cron job running rsync periodically or on-demand to keep the server up-to-date. And similarly updating the library on the main machine from the server in case of mobile uploads.

use case 2:
share the same photos volume with another docker container that could change the files in the library (e.g. another photo library app or something completely different that works with photos)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This'll be covered by #12293

**Reply by comebacktocode** [reply]
hey @bo0tzz what happens today, before #12293 gets implemented, when I do manual changes directly at the /library/ content? I did the mistake of changing files and folders there directly, because the storage template results made no sense. Did I break things? Should I reimport the content again?

thanks

**Reply by bo0tzz** [reply]
Yes, that will break things. There's no easy way to tell exactly what needs fixing, so the most straightforward path is reimporting.
